lib/src/database/matrix_sdk_database.dart

```diff
@@ -0,0 +1,1590 @@
+        await _preloadRoomStateBox.clear();
+        await _nonPreloadRoomStateBox.clear();
+        await _roomMembersBox.clear();
+        await _eventsBox.clear();
+        await _timelineFragmentsBox.clear();
+        await _outboundGroupSessionsBox.clear();
+        await _presencesBox.clear();
+        await _clientBox.delete('prev_batch');
+      });
+
+  @override
+  Future<void> clearSSSSCache() => _ssssCacheBox.clear();
+
+  @override
+  Future<void> close() async => _collection.close();
+
+  @override
+  Future<void> deleteFromToDeviceQueue(int id) async {
+    await _toDeviceQueueBox.delete(id.toString());
+    return;
+  }
+
+  @override
+  Future<void> deleteOldFiles(int savedAt) async {
+    final dir = fileStoragePath;
+    final deleteFilesAfterDuration = this.deleteFilesAfterDuration;
+    if (!supportsFileStoring ||
+        dir == null ||
+        deleteFilesAfterDuration == null) {
+      return;
+    }
+    final entities = await dir.list().toList();
+    for (final file in entities) {
+      if (file is! File) continue;
+      final stat = await file.stat();
+      if (DateTime.now().difference(stat.modified) > deleteFilesAfterDuration) {
+        Logs().v('Delete old file', file.path);
+        await file.delete();
+      }
+    }
+  }
+
+  @override
+  Future<void> forgetRoom(String roomId) => transaction(() async {
+        await _timelineFragmentsBox.delete(TupleKey(roomId, '').toString());
+        final eventsBoxKeys = await _eventsBox.getAllKeys();
+        for (final key in eventsBoxKeys) {
+          final multiKey = TupleKey.fromString(key);
+          if (multiKey.parts.first != roomId) continue;
+          await _eventsBox.delete(key);
+        }
+        final preloadRoomStateBoxKeys = await _preloadRoomStateBox.getAllKeys();
+        for (final key in preloadRoomStateBoxKeys) {
+          final multiKey = TupleKey.fromString(key);
+          if (multiKey.parts.first != roomId) continue;
+          await _preloadRoomStateBox.delete(key);
+        }
+        final nonPreloadRoomStateBoxKeys =
+            await _nonPreloadRoomStateBox.getAllKeys();
+        for (final key in nonPreloadRoomStateBoxKeys) {
+          final multiKey = TupleKey.fromString(key);
+          if (multiKey.parts.first != roomId) continue;
+          await _nonPreloadRoomStateBox.delete(key);
+        }
+        final roomMembersBoxKeys = await _roomMembersBox.getAllKeys();
+        for (final key in roomMembersBoxKeys) {
+          final multiKey = TupleKey.fromString(key);
+          if (multiKey.parts.first != roomId) continue;
+          await _roomMembersBox.delete(key);
+        }
+        final roomAccountDataBoxKeys = await _roomAccountDataBox.getAllKeys();
+        for (final key in roomAccountDataBoxKeys) {
+          final multiKey = TupleKey.fromString(key);
+          if (multiKey.parts.first != roomId) continue;
+          await _roomAccountDataBox.delete(key);
+        }
+        await _roomsBox.delete(roomId);
```

Comment:
This is one of the cases, that would be a lot more efficient without the boxes. However, if you type the database key, then possibly we could make this a lot more efficient as well, since we could expose a "delete by partial key" function in the future (and add an index on the first partial key).

Reply:
See above